src/widgets/AppBar.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from "react";
+import { StackHeaderProps } from "@react-navigation/stack";
+import { Appbar } from "react-native-paper";
+
+import MenuButton from "./MenuButton";
+
+export default function AppBar(props: StackHeaderProps & { menuFallback: boolean }) {
+  const { insets, menuFallback, navigation, previous, scene } = props;
+  const { options } = scene.descriptor;
+  const title = (options.headerTitle)
+    ? options.headerTitle
+    : (options.title)
+      ? options.title
+      : scene.route.name;
+  const left = (options.headerLeft)
```

Comment:
This looks extremely complex. I realise you probably copied it from somewhere, though it looks very error prone.

Reply:
Should probably be simplified and maybe split up. Or at the very least, add parenthesis around each inline if.

Reply:
I copied the logic behind `title`, except this one is with React Components. That's actually what makes the main behavior we want : show the headerLeft option if available, if not and we can go back, show the back button, if not and we have menuFallback show the menu, otherwise don't show anything.

I already put parenthesis around each condition, like you asked me to do before. Do you want me to nest the conditions inside parenthesis? I thought the indentation showed well new conditions. Maybe I could just use plain `if… else`, it would be easier to read.

Reply:
Yeah, that's what I meant this time:
```
(options.headerTitle) ? options.headerTitle : ((options.title) ? options.title : scene.route.name)
```
I showed the example for the wrong line, but you see what I mean. Yeah, potentially an if..else could be better.